backends/bmv2/psa_switch/psaSwitch.cpp

```diff
@@ -116,6 +117,24 @@ void PsaProgramStructure::createTypes(ConversionContext* ctxt) {
     // add enums to json
 }
 
+void PsaProgramStructure::createScalars(ConversionContext* ctxt) {
+    auto name = scalars.begin()->first;
+    ctxt->json->add_header("scalars_t", name);
+    ctxt->json->add_header_type("scalars_t");
+
+    for (auto kv : scalars) {
+        LOG5("Adding a scalar field " << kv.second << " to generated json");
+        auto field = new Util::JsonArray();
+        auto ftype = typeMap->getType(kv.second, true);
+        if (auto type = ftype->to<IR::Type_Bits>()) {
```

Comment:
What if it's not?

Reply:
I use this function to handle all statements that is in the format `bit<x> varName;`, thus I believe all targeted statements will go in the if block.

If a statement does not go into the if block, the statement is not of format `bit<x> varName;` and should not be caught by this function.